File: mine/src/main/java/com/klcw/app/mine/util/MineUtils.java
package com.klcw.app.mine.util;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.text.TextUtils;

import com.billy.cc.core.component.CC;
import com.klcw.app.lib.widget.BLToast;
import com.klcw.app.mine.activity.MineActivity;
import com.klcw.app.mine.constant.MineConstant;

/**
 * @author kk
 * @datetime: 2018/10/24
 * @desc:
 */
public class MineUtils {

    /**
     * 跳转MineActivity
     *
     * @param cc 参数
     */
    public static void openMineActivity(CC cc) {
        String param = cc.getParamItem(MineConstant.KRY_PARAM);
        if (TextUtils.isEmpty(param)) {
            BLToast.showToast(cc.getContext(), "参数不能为空");
            return;
        }
        Context context = cc.getContext();
        Intent intent = new Intent(context, MineActivity.class);
        if (!(context instanceof Activity)) {
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        }
        intent.putExtra(MineConstant.KRY_PARAM, param);
        context.startActivity(intent);
    }

}

File: mine/src/main/java/com/kk/app/mine/util/MineUtils.kt
package com.kk.app.mine.util

import android.app.Activity
import android.content.Intent
import android.text.TextUtils
import com.billy.cc.core.component.CC
import com.kk.app.lib.widget.BLToast
import com.kk.app.mine.activity.MineActivity
import com.kk.app.mine.constant.MineConstant

/**
 * @author kk
 * @datetime: 2018/10/24
 * @desc:
 */
object MineUtils {
    /**
     * 跳转MineActivity
     *
     * @param cc 参数
     */
    fun openMineActivity(cc: CC) {
        val param = cc.getParamItem<String>(MineConstant.Companion.KRY_PARAM)
        if (TextUtils.isEmpty(param)) {
            BLToast.showToast(cc.context, "参数不能为空")
            return
        }
        val context = cc.context
        val intent = Intent(context, MineActivity::class.java)
        if (context !is Activity) {
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        }
        intent.putExtra(MineConstant.Companion.KRY_PARAM, param)
        context.startActivity(intent)
    }
}